lib/ansible/modules/packaging/os/flatpak_remote.py

```diff
@@ -0,0 +1,245 @@
+
+DOCUMENTATION = r'''
+---
+module: flatpak_remote
+version_added: '2.6'
+requirements:
+- flatpak
+author:
+- John Kwiatkoski (@jaykayy)
+- Alexander Bethke (@oolongbrothers)
+short_description: Manage flatpak repository remotes
+description:
+- Manage flatpak repository remotes.
+options:
+  name:
+    description:
+    - When I(state) is set to C(present), I(name) is added as a remote for installing flatpaks.
+      When used with I(state=absent) the remote with thet name will be removed.
+    required: true
+  remote:
+    description:
+    - When I(state) is set to C(present), I(remote) url is added as a flatpak remote for the
+      specified installation C(method).
+      When used with I(state=absent), this is not required.
+    required: false
+  method:
+    description:
+    - Determines the type of installation to work on. Can be C(user) or C(system) installations.
+    choices: [ user, system ]
+    required: false
+    default: system
+  executable:
+    description:
+    - The path to the C(flatpak) executable to use. The default will look for
+      the c(flatpak) executable on the path
+    default: flatpak
+  state:
+    description:
+      - Set to C(present) will install the flatpak remote.
+      - Set to C(absent) will remove the flatpak remote.
+    choices: [ absent, present ]
+    default: present
+'''
+
+EXAMPLES = r'''
+- name: Add the Gnome flatpak remote to the system installation under the name 'gnome'.
+  flatpak_remote:
+    name: gnome
+    state: present
+    remote: https://sdk.gnome.org/gnome-apps.flatpakrepo
+    method: system
+
+- name: Remove the Gnome flatpak remote  from the user installation.
+  flatpak_remote:
+    name: gnome
+    state: absent
+
+- name: Add the flathub flatpak repository remote to the user installation.
+  flatpak_remote:
+    name: flathub
+    state: present
+    remote:  https://dl.flathub.org/repo/flathub.flatpakrepo
+    method: user
+
+- name: Remove the flathub flatpak repository remote from the system installtion.
+  flatpak_remote:
+    name: flathub
+    state: absent
+    method: system
+'''
+
+RETURN = r'''
+reason:
+  description: On failure, the output for the failure
+  returned: failed
+  type: string
+  sample: error while installing...
+name:
+  description: Remote of flatpak given for the operation
+  returned: always
+  type: string
+  sample: https://sdk.gnome.org/gnome-apps.flatpakrepo
+'''
+
+import subprocess
+from ansible.module_utils.basic import AnsibleModule
+
+
+def add_remote(module, binary, name, remote, method):
+    """
+    Add a new remote.
+
+    returns:
+        result, type: int
+            The result status of the operation
+            Possible values:
+            0 - operation successful
+            1 - operation failed
+        output, type: str
+            Output of the operation
+            Especially helpful to know that's wrong when the operation failed
+    """
+    if module.check_mode:
```

Comment:
IMO it is better to move this test further up in **_flatpak_command()**.
And leave the module through the final **exit_json()** just as if you would if you're not in check_mode.